### PR TITLE
feat(CORE-1238): Add upgrade guide from EDR to pipelines v2

### PR DESCRIPTION
**Pull request by MoonMoon1919**

### CODE REVIEW

File: _docs-sources/pipelines/upgrading/index.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+- `PIPELINES_DISPATCH_TOKEN`: Fine Grained token that only has workflow permissions to execute workflows in the `infrastructure-pipelines` repository. This token should be associated with your existing CI user.
+- `INFRA_LIVE_ACCESS_TOKEN`: Fine grained token with READ and WRITE access to your `infrastructure-live` repository. This token should be associated with your existing CI user.
+
+## Creating your infrastructure-pipelines repository
+
+Modern Gruntwork Pipelines employ a dual-repository approach to isolate infrastructure as code (IaC) code definitions and IaC deployment mechanisms. This approach signifies that a single, highly secured repository, with write access limited to a specific group of administrators, contains both the code responsible for deploying your IaC and the access to your AWS accounts. Workflows within this repository leverage OpenId Connect (OIDC) and AWS IAM roles for creating short-lived session credentials, as opposed to storing long-lived credentials as secrets.
```
Comment (ellisonc) [inline _docs-sources/pipelines/upgrading/index.md:32]: "employ" here implies that `Modern Gruntwork Pipelines` is plural, in other places in this document it's interpreted as a singular product name. See line 16 "Gruntwork Pipelines uses". We should be consistent. I'm more inclined to treat it as a singular product name, but happy to be overruled.

File: _docs-sources/pipelines/upgrading/index.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+## Updating your accounts file
+
+As a part of separate product improvement the `accounts.json` file has been changed to `accounts.yml`. In addition to readability improvements, we found that Terraform's built-in YAML formatters better supported our use case of re-generating this file to support adding new accounts using GitOps workflows. There is no functional difference between the two different file types in the EDR version of pipelines or Modern Gruntwork Pipelines.
+
+As part of a separate product improvement initiative, we have transitioned from the `accounts.json` file to `accounts.yml`. This change not only enhances readability but is also functionally better, with Terraform's built-in YAML formatter better meeting our requirement for regenerating a formatted file to facilitate the addition of new accounts through GitOps workflows. It's important to note that there is no functional distinction between these two file types in either the EDR version of pipelines or Modern Gruntwork Pipelines.
```
Comment (ellisonc) [inline _docs-sources/pipelines/upgrading/index.md:114]: this line is mostly duplicated from above

File: _docs-sources/pipelines/upgrading/index.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
+# Upgrading from the ECS Deploy Runner
+
+This migration guide is intended for those running the ECS Deploy Runner (EDR) based Gruntwork Pipelines upgrading to Modern Gruntwork Pipelines. Upgrading can be done using Pipelines itself in a series of steps using code generation and pull requests.
```
Comment (ellisonc) [inline _docs-sources/pipelines/upgrading/index.md:3]: "Modern Gruntwork Pipelines" gets clunky. The first time we mention it up here we should say something like:

Modern Gruntwork Pipelines (Referred to as Gruntwork Pipelines or simply Pipelines in the rest of this article. The prior version will be explicitly called EDR pipelines for the remainder of this article.)

I don't feel super strongly about this though, so pls ignore if you want, I just got tired of reading extra words.

File: _docs-sources/pipelines/upgrading/index.md (hunk 1 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/upgrading/index.md:32]: Ah good catch, thank you!

File: _docs-sources/pipelines/upgrading/index.md (hunk 3 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/upgrading/index.md:3]: Good call, it used to say `v2` then i did a find a replace. Your idea is better!

File: _docs-sources/pipelines/upgrading/index.md (hunk 1 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/upgrading/index.md]: Fixed in [73295b2](https://github.com/gruntwork-io/docs/pull/1115/commits/73295b26e2fbe7ab38f0addf4381fb0d3a67be12)

File: _docs-sources/pipelines/upgrading/index.md (hunk 2 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/upgrading/index.md]: Fixed in [73295b2](https://github.com/gruntwork-io/docs/pull/1115/commits/73295b26e2fbe7ab38f0addf4381fb0d3a67be12)

File: _docs-sources/pipelines/upgrading/index.md (hunk 3 above)
Comment (MoonMoon1919) [reply _docs-sources/pipelines/upgrading/index.md]: Fixed in [73295b2](https://github.com/gruntwork-io/docs/pull/1115/commits/73295b26e2fbe7ab38f0addf4381fb0d3a67be12)

**Review by ellisonc — APPROVED** [review]

**Comment by josh-padnick** [comment]
@MoonMoon1919 This guide is fantastic! Nicely done. 👍 I added some minor tweaks here, but in my opinion this is ready to merge.

**Review by josh-padnick — APPROVED** [review]

**Pull request merged**